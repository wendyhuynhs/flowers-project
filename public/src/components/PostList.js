import React, { useEffect } from "react";
import { connect } from "react-redux";
import styles from "../styles/postList.scss";
import Post from "./Post.js";

const PostList = ({ posts, getPosts }) => {
  useEffect(() => {
    getPosts();
  }, []);

  return (
    <div>
      {posts.length > 1 &&
        posts.map((post) => {
          return (
            <div key={post.id}>
              <Post post={post} />
            </div>
          );
        })}

      {posts[0] === "error" && (
        <div className={styles.error}>Error Loading Posts</div>
      )}
    </div>
  );
};

const mapStateToProps = ({ posts }) => {
  return {
    posts,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getPosts: () => dispatch({ type: "GET_POSTS" }),
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(PostList);
