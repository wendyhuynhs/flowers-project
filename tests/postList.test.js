import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";

import PostList from "../public/src/components/PostList";

const mockStore = configureMockStore([]);

describe("Post List Component", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      posts: [],
    });
    store.clearActions();

    component = renderer.create(
      <Provider store={store}>
        <PostList />
      </Provider>
    );
  });

  it(" should render state from store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
