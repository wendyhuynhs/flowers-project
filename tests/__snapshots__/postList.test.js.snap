// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post List Component  should render state from store 1`] = `<div />`;
